pygac/gac_io.py

```diff
@@ -192,53 +201,22 @@ def save_gac(satellite_name,
     # Earth-Sun distance correction factor
     corr = 1.0 - 0.0334 * np.cos(2.0 * np.pi * (jday - 2) / 365.25)
 
-    # Slice scanline range requested by user
-    ref1 = ref1[start_line:end_line+1, :].copy()
-    ref2 = ref2[start_line:end_line+1, :].copy()
-    ref3 = ref3[start_line:end_line+1, :].copy()
-    bt3 = bt3[start_line:end_line+1, :].copy()
-    bt4 = bt4[start_line:end_line+1, :].copy()
-    bt5 = bt5[start_line:end_line+1, :].copy()
-    sun_zen = sun_zen[start_line:end_line+1, :].copy()
-    sun_azi = sun_azi[start_line:end_line+1, :].copy()
-    sat_zen = sat_zen[start_line:end_line+1, :].copy()
-    sat_azi = sat_azi[start_line:end_line+1, :].copy()
-    rel_azi = rel_azi[start_line:end_line+1, :].copy()
-    lats = lats[start_line:end_line+1, :].copy()
-    lons = lons[start_line:end_line+1, :].copy()
-    qual_flags = qual_flags[start_line:end_line+1, :].copy()
-    xutcs = xutcs[start_line:end_line+1].copy()
-
-    if midnight_scanline is not None:
-        # Update midnight scanline to the final scanline range
-        midnight_scanline -= (temp_start_line + start_line)
-
-        # Set midnight scanline to None if it has been removed due to invalid
-        # lat/lon info (< 0) or lies outside the user defined scanline range
-        if midnight_scanline < 0 or (midnight_scanline > end_line or
-                                     midnight_scanline < start_line):
-            midnight_scanline = None
-
-    # Compute total number of scanlines
-    total_number_of_scan_lines = end_line - start_line + 1
-
-    # Correct for temporary scan motor issue.
-    #
-    # TODO: The thresholds in tsm.flag_pixels() were derived from the final
-    #       pygac output, that's why the correction is applied here. It would
-    #       certainly be more consistent to apply the correction in GACReader,
-    #       but that requires a new threshold analysis.
-    if tsmcorr:
-        LOG.info('Correcting for temporary scan motor issue')
-        tic = datetime.datetime.now()
-        (ref1, ref2, bt3, bt4, bt5, ref3) = flag_tsm_pixels(channel1=ref1,
-                                                            channel2=ref2,
-                                                            channel3b=bt3,
-                                                            channel4=bt4,
-                                                            channel5=bt5,
-                                                            channel3a=ref3,
-                                                            fillv=MISSING_DATA)
-        LOG.debug('TSM correction took: %s', str(datetime.datetime.now() - tic))
+    # Apply scaling & offset
+    bt3 -= 273.15
+    bt4 -= 273.15
+    bt5 -= 273.15
+    for array in [bt3, bt4, bt5, ref1, ref2, ref3, sun_zen, sat_zen, sun_azi,
+                  sat_azi, rel_azi]:
+        array *= 100
+    for array in [lats, lons]:
+        array *= 1000.0
```

Comment:
just for esthetics :) why 100 vs 1000.0 ? why not both ints or floats ?

Reply:
I'll make it both floats :nail_care: